File: Frontend/trackr-administrador/src/app/shared/http/catalogo/giro-comercial.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';

import { GiroComercial } from '@models/catalogo/giro-comercial';

@Injectable()
export class GiroComercialService {
  private dataUrl = 'giroComercial/';

  constructor(private http: HttpClient) { }

  public consultarTodos(): Observable<GiroComercial[]> {
    return this.http.get<GiroComercial[]>(this.dataUrl + 'consultarTodos');
  }

  public consultar(idGiroComercial: number): Observable<GiroComercial> {
    return this.http.get<GiroComercial>(this.dataUrl + `consultar/${idGiroComercial}`);
  }

  public agregar(tipoConcepto: GiroComercial): Observable<void> {
    return this.http.post<void>(this.dataUrl + 'agregar', tipoConcepto);
  }

  public editar(tipoConcepto: GiroComercial): Observable<void> {
    return this.http.put<void>(this.dataUrl + 'editar', tipoConcepto);
  }

  public eliminar(idGiroComercial: number): Observable<void> {
    return this.http.delete<void>(this.dataUrl + `eliminar/${idGiroComercial}`);
  }
}

File: Backend/TrackrAPI/TrackrAPI/Services/Catalogo/GiroComercialService.cs
using TrackrAPI.Models;
using TrackrAPI.Repositorys.Catalogo;
using System.Collections.Generic;
using System.Linq;

namespace TrackrAPI.Services.Catalogo
{
    public class GiroComercialService
    {
        private IGiroComercialRepository giroComercialRepository;
        private GiroComercialValidatorService giroComercialValidatorService;

        public GiroComercialService(
            IGiroComercialRepository giroComercialRepository,
            GiroComercialValidatorService giroComercialValidatorService
        )
        {
            this.giroComercialRepository = giroComercialRepository;
            this.giroComercialValidatorService = giroComercialValidatorService;
        }

        public IEnumerable<GiroComercial> ConsultarTodos()
        {
            return giroComercialRepository.ConsultarTodos();
        }

        public GiroComercial Consultar(int idGiroComercial)
        {
            return giroComercialRepository.Consultar(idGiroComercial);
        }

        public int Agregar(GiroComercial giroComercial)
        {
            giroComercialValidatorService.ValidarAgregar(giroComercial);
            giroComercialRepository.Agregar(giroComercial);
            return giroComercial.IdGiroComercial;
        }

        public void Editar(GiroComercial giroComercial)
        {
            giroComercialValidatorService.ValidarEditar(giroComercial);
            giroComercialRepository.Editar(giroComercial);
        }

        public void Eliminar(int idGiroComercial)
        {
            giroComercialValidatorService.ValidarEliminar(idGiroComercial);

            GiroComercial giroComercial = giroComercialRepository.Consultar(idGiroComercial);
            giroComercialRepository.Eliminar(giroComercial);
        }
    }
}
